### PR TITLE
Move structured field into its own module under tmt.utils

### PR DESCRIPTION
**Pull request by happz**
Another round of tmt.utils liberations.

Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

**Review by thrix — APPROVED** [review]
LGTM

**Review by therazix — APPROVED** [review]

File: tmt/utils/structured_field.py [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+
+import fmf.utils
+
+from tmt.utils import StructuredFieldError, format_value, pure_ascii
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#  StructuredField
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# TODO: deprecate ASAP
```
Comment (martinhoyer) [inline tmt/utils/structured_field.py:14]: I take it this is new TODO - maybe extend the comment with why it's being used?

**Review by martinhoyer — APPROVED** [review]
Feels wrong to make such suggestions to @happz, but could you next time split it to two commits, one where the code was just moved to another file and other with any potential changes, so it's easier to review?

File: tmt/utils/structured_field.py (hunk 1 above)
Comment (happz) [reply tmt/utils/structured_field.py:14]: Yeah, that's a leftover: I tried to get this new file covered with pyright, but there were so many changes that it turned exactly into a mix of moving files and making changes to them. So I undid all my changes - at least I thought so, this one slipped through my review.

**Comment by happz** [comment]
> Feels wrong to make such suggestions to @happz, but could you next time split it to two commits, one where the code was just moved to another file and other with any potential changes, so it's easier to review?

I hope that once the comment WRT `TODO` is addressed, this would be the actual state of things. Only moves, no changes beyond what's needed to move the code & make it work in new place.

**Comment by martinhoyer** [comment]
> > Feels wrong to make such suggestions to @happz, but could you next time split it to two commits, one where the code was just moved to another file and other with any potential changes, so it's easier to review?
> 
> I hope that once the comment WRT `TODO` is addressed, this would be the actual state of things. Only moves, no changes beyond what's needed to move the code & make it work in new place.

Thanks. In that case I'd rephrase the suggestion to "mention that *foo* was just moved, not changed for easier reviews" :)

File: tmt/utils/structured_field.py (hunk 1 above)
Comment (happz) [reply tmt/utils/structured_field.py]: Dropped in 2666b278

**Review by seberm — APPROVED** [review]

**Review by KwisatzHaderach — APPROVED** [review]
